app_spec/zomes/blog/code/src/blog.rs

```diff
@@ -107,17 +116,24 @@ pub fn handle_post_address(content: String) -> ZomeApiResult<Address> {
     hdk::entry_address(&post_entry(content))
 }
 
+pub static BOB_AGENT_ID: &'static str =
+    "HcScj5GbxXdTq69sfnz3jcA4u5f35zftsuu5Eb3dBxHjgd9byUUW6JmN3Bvzqqr";
+
 fn is_my_friend(addr: Address) -> bool {
-    // this is "alice's" hash
-    addr == Address::from("HcScjwO9ji9633ZYxa6IYubHJHW6ctfoufv5eq4F7ZOxay8wR76FP4xeG9pY3ui")
+    addr == Address::from(BOB_AGENT_ID)
 }
 
 pub fn handle_request_post_grant() -> ZomeApiResult<Option<Address>> {
-    let addr = CAPABILITY_REQ.provenance.source();
+    // if it worked for the nodejs conductor to make calls with different provenances
+    // we could get the caller we want from the CAPABILITY_REQ global like this:
+    //    let addr = CAPABILITY_REQ.provenance.source();
+    // but it doesn't work yet so for this test we are hard-coding the "friend"" to bob
```

Comment:
This is an open conversation to loop back to. Last time we talked about it, it sounded like it was beneficial to test it the way you did here, because that would be more like how a real app would work.

Reply:
yep. updated the comment a bit